crates/weaver_health/src/attribute_advice.rs

```diff
@@ -0,0 +1,382 @@
+                key: "deprecated".to_owned(),
+                value: match deprecated {
+                    Deprecated::Renamed { .. } => Value::String("renamed".to_owned()),
+                    Deprecated::Obsoleted { .. } => Value::String("obsoleted".to_owned()),
+                    Deprecated::Uncategorized { .. } => Value::String("uncategorized".to_owned()),
+                },
+                message: deprecated.to_string(),
+                advisory: Advisory::Violation,
+            })
+        } else {
+            None
+        }
+    }
+}
+
+/// An advisor that checks if an attribute is stable from the stability field in the semantic convention
+/// The value will be the stability level
+pub struct StabilityAdvisor;
+// TODO: Configurable Advisory level, strictly stable would mean Violation
+
+impl Advisor for StabilityAdvisor {
+    fn advise(
+        &self,
+        _attribute: &SampleAttribute,
+        _health_checker: &AttributeHealthChecker,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Option<Advice> {
+        if let Some(attribute) = semconv_attribute {
+            match attribute.stability {
+                Some(ref stability) if *stability != Stability::Stable => Some(Advice {
+                    key: "stability".to_owned(),
+                    value: Value::String(stability.to_string()),
+                    message: "Is not stable".to_owned(),
+                    advisory: Advisory::Improvement,
+                }),
+                _ => None,
+            }
+        } else {
+            None
+        }
+    }
+}
+
+/// An advisor that checks if an attribute matches name formatting rules
+pub struct NameFormatAdvisor {
+    regex: Regex,
+}
+impl NameFormatAdvisor {
+    #[must_use]
+    /// Create a new NameFormatAdvisor
+    pub fn new(pattern: &str) -> Self {
+        NameFormatAdvisor {
+            regex: Regex::new(pattern).expect("regex pattern must be valid"),
+        }
+    }
+}
+impl Default for NameFormatAdvisor {
+    fn default() -> Self {
+        Self::new(r"^[a-z][a-z0-9]*([._][a-z0-9]+)*$")
+    }
+}
+
+impl Advisor for NameFormatAdvisor {
+    fn advise(
+        &self,
+        attribute: &SampleAttribute,
+        _health_checker: &AttributeHealthChecker,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Option<Advice> {
+        // Don't provide advice if the attribute is a match
+        if semconv_attribute.is_some() {
+            return None;
+        }
+
+        if !self.regex.is_match(&attribute.name) {
+            Some(Advice {
+                key: "invalid_format".to_owned(),
+                value: Value::String(attribute.name.clone()),
+                message: "Does not match name formatting rules".to_owned(),
+                advisory: Advisory::Violation,
+            })
+        } else {
+            None
+        }
+    }
+}
+
+/// An advisor that provides advice on the namespace of an attribute
+pub struct NamespaceAdvisor {
```

Comment:
Similar question here.

Reply:
Same answer as above. This was also taken from my own tool. For my developers it has proven useful to see, for example, that a sample attribute extends an existing namespace. This may or may not be intentional so this is a good point of guidance. Likewise, indicating that the sample would collide with a defined attribute or that it's missing a namespace altogether is all just useful guidance. Again, whether you decide to formally define the attribute or not.

I will move this to rego too.

Reply:
Moved to rego in this commit: [f6b7023](https://github.com/open-telemetry/weaver/pull/630/commits/f6b702395e8c346eafa7626538e5e06b634eddfb)